public-ui/src/app/data-browser/views/survey-chart/survey-chart-react.component.tsx

```diff
@@ -0,0 +1,179 @@
+        + q.actualQuestionNumber + ' - ' + q.conceptName + ' - ' + answer.stratum4 +
+        ' - ' + g, searchTerm, null);
+    }
+    triggerEvent('conceptClick', 'View Graphs',
+      'Expand to see graphs', surveyName + ' - Q'
+      + q.actualQuestionNumber + ' - ' + q.conceptName + ' - ' + answer.stratum4 +
+      ' - ' + g, searchTerm, null);
+    q.graphToShow = g;
+    switch (g) {
+        case GraphType.AgeWhenSurveyWasTaken:
+            q.selectedAnalysis = q.ageAnalysis;
+            break;
+        case GraphType.SurveyVersion:
+            q.selectedAnalysis = q.versionAnalysis;
+            break;
+        default:
+            q.selectedAnalysis = q.genderAnalysis;
+            break;
+    }
+    this.setState({graphToShow: g, selectedChartAnalysis: q.selectedAnalysis, isLoaded: true,
+        displayGraphErrorMessage: q.selectedAnalysis === undefined ||
+        (q.selectedAnalysis && q.selectedAnalysis.results.filter(a => a.stratum3 ===
+        answer.stratum3).length === 0) });
+    triggerEvent('graphTabClick', 'Survey Graph',
+      'Click', surveyName + ' - ' + q.graphToShow + ' - Q'
+      + q.actualQuestionNumber + ' - ' + q.conceptName + ' - ' + answer.stratum4 +
+      ' - ' + g, searchTerm, null);
+  }
+
+  getTooltipKey(g: string) {
+      if (g === 'Sex Assigned at Birth') {
+        return 'surveyBSChart';
+      } else if (g === 'Survey Versions') {
+        return 'versionChartHelpText';
+      } else if (g === 'Age When Survey Was Taken') {
+        return 'surveyAgeChartHelpText';
+      } else {
+        return g;
+      }
+  }
+
+  getLabel(g) {
+    const {surveyName, question, answer} = this.props;
+    return surveyName + ' - Q' +  question.actualQuestionNumber + ' - ' +
+    question.conceptName + ' - ' + answer.stratum4 + ' - ' + g;
+  }
+
+
+  render() {
+      const {graphButtons, question, answer, searchTerm, surveyCountAnalysis, selectedResult, versionAnalysis} = this.props;
+      const { isLoaded, graphToShow, displayGraphErrorMessage, selectedChartAnalysis } = this.state;
+      const tabIndex = 0;
+      return <React.Fragment>
+        <div className='survey-graph-menu'>
+            {
+              graphButtons.map((g, index) => {
+                return (
+                 <div onClick={() => this.selectGraphType(g, question, answer)}
+                 className={graphToShow === g ? 'active survey-chart-choice' : 'survey-chart-choice'}
+                 tabIndex={tabIndex} key={index}>
+                 <span>{g}</span>
+                 <TooltipReactComponent tooltipKey={this.getTooltipKey(g)}
+                 label={this.getLabel(g)} searchTerm={searchTerm} action='Survey Chart Tooltip'>
+                 </TooltipReactComponent>
+                 </div>
+                 );
+              })
+            }
+        </div>
+{displayGraphErrorMessage
+    ? <div className='graph-error-message'>
+                  <ErrorMessageReactComponent dataType='chart'></ErrorMessageReactComponent>
```

Comment:
Use self-closing syntax if there's no nested content:
```suggestion
                  <ErrorMessageReactComponent dataType='chart'/>
```

Reply:
changed thanks.